CMakeLists.txt

```diff
@@ -57,16 +57,33 @@ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
+# Conan dependencies
+include(conan)
+set(DYNAMIC_LIB_TRUE_FALSE "False")
+if(${DYNAMIC_LIB})
+  set(DYNAMIC_LIB_TRUE_FALSE "True")
+endif()
+
+# Detect compiler, build type etc from cmake settings
+conan_cmake_autodetect(conan_settings)
+conan_cmake_run(REQUIRES eigen/3.3.9
+                         gtest/1.11.0
+                         benchmark/1.5.3
+        OPTIONS gtest:shared=${DYNAMIC_LIB_TRUE_FALSE}
+                benchmark:shared=${DYNAMIC_LIB_TRUE_FALSE}
+        SETTINGS ${conan_settings}
+        BASIC_SETUP CMAKE_TARGETS
+        BUILD outdated
+        KEEP_RPATHS)
```

Comment:
This goes through a few steps to get hold of the dependencies:
- checks if you already have the requested dependencies in your local conan cache (`~/.conan/data` on Linux) and if they are compatible with your current build env. If not...
- looks through your configured conan "remotes" (remote conan repo servers) looking for a binary for each library that is compatible with your build env. Downloads them if found. If not...
-  looks through your configured conan "remotes" for a [conan recipe](https://docs.conan.io/en/latest/creating_packages/getting_started.html#creating-the-package-recipe) for dependency and uses it to download the dependency's source and build it.

All three dependencies are available in the main public remote "Conan Center", which is configured by default when you install Conan.